app/src/main/java/org/stepic/droid/code/ui/CodeAnalyzer.kt

```diff
@@ -141,4 +146,21 @@ object CodeAnalyzer {
     }
 
     fun getBracketsPair(bracket: String) = brackets.entries.find { it.key == bracket || it.value == bracket }
+
+
+    fun resolveAutocomplete(cursorPosition: Int, lang: String, text: String): AutocompleteState {
+        val next = getNextSymbol(cursorPosition, text)
+        if (next == null || Character.isWhitespace(next[0])) {
+            val prefix = text.takeLastFromIndexWhile(cursorPosition) { // get token before cursor
```

Comment:
we should check only `MIN_AUTOCOMPLETE_LENGTH` chars, not all symbols between cursor and whitespace. Performance can be low if prefix is too long

Reply:
We need all `prefix` anyway to find suggestions.